packages/terra-popup/src/_PopupContent.jsx

```diff
@@ -0,0 +1,286 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import onClickOutside from 'react-onclickoutside';
+import ContentContainer from 'terra-content-container';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import FocusTrap from 'focus-trap-react';
+import './PopupContent.scss';
+
+/**
+ * Directional classes to be applied by a presenting component.
+ */
+const CONTENT_CLASSES = {
+  top: 'terra-PopupContent--arrowTop',
+  bottom: 'terra-PopupContent--arrowBottom',
+  left: 'terra-PopupContent--arrowLeft',
+  right: 'terra-PopupContent--arrowRight',
+};
+
+/**
+ * Mirrored directional classes, used to flip the arrow on repositioning.
+ */
+const CONTENT_OPPOSITE_CLASSES = {
+  top: 'terra-PopupContent--arrowBottom',
+  bottom: 'terra-PopupContent--arrowTop',
+  left: 'terra-PopupContent--arrowRight',
+  right: 'terra-PopupContent--arrowLeft',
+};
+
+/**
+ * Key code value for the escape key.
+ */
+const KEYCODES = {
+  ESCAPE: 27,
+};
+
+/**
+ * Valid positions for the arrow, applies a margin to the corresponding side.
+ */
+const ARROW_POSITIONS = [
+  'top',
+  'bottom',
+  'left',
+  'right',
+];
+
+/**
+ * Margin value used for calculations.
+ */
+const POPUP_MARGIN = 9;
+
+const propTypes = {
+  /**
+   * The arrow to be placed within the popup frame.
+   */
+  arrow: PropTypes.element,
+  /**
+   * The initial arrow position.
+   */
+  arrowPosition: PropTypes.oneOf(ARROW_POSITIONS),
+  /**
+   * The children to be presented as the popup's content.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * CSS classnames that are append to the popup content body.
+   */
+  classNameInner: PropTypes.string,
+  /**
+   * Whether or not the using the escape key should trigger the onRequestClose callback.
+   */
+  closeOnEsc: PropTypes.bool,
```

Comment:
Is it possible to use this convention for the bools in this file? https://github.cerner.com/terra/conventions/blob/master/react.md#for-boolean-props-define-defaultprop-value-as-false

Someline like isClosableOnEsc should work